### PR TITLE
[draft] Fix keepalive initialization in awss3 input

### PR DESCRIPTION
**Pull request by faec**
## Proposed commit message

The `visibility_timeout` flag in the SQS input isn't initialized correctly. It is used when extending visibility timeouts, but not when initially fetching the message, so the first timeout defaults to the global queue setting (which is configured by the user when creating the SQS queue). If the client-side timeout is higher than the AWS-side timeout, then Agent will not extend the visibility in time, and any objects that aren't processed by the AWS-side timeout will be returned to the queue.

The intention is for the visibility timeout to be applied regardless of the SQS-side defaults. The solution is to call the SQS `ChangeMessageVisibility` API at the _beginning_ of the keepalive helper, rather than waiting until the first timeout expires.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: x-pack/filebeat/input/awss3/sqs_s3_event.go [hunk 1]
```diff
@@ -211,19 +226,6 @@ func (p *sqsS3EventProcessor) keepalive(ctx context.Context, log *logp.Logger, w
 				"expires_at", time.Now().UTC().Add(p.sqsVisibilityTimeout))
 			p.metrics.sqsVisibilityTimeoutExtensionsTotal.Inc()
 
-			// Renew visibility.
```
Comment (kaiyan-sheng) [inline x-pack/filebeat/input/awss3/sqs_s3_event.go:214]: should we keep the renew part here to change the visibility timeout?

File: x-pack/filebeat/input/awss3/sqs_s3_event.go [hunk 2]
```diff
@@ -202,6 +202,21 @@ func (p *sqsS3EventProcessor) keepalive(ctx context.Context, log *logp.Logger, w
 	defer t.Stop()
 
 	for {
+		// Renew visibility.
```
Comment (kaiyan-sheng) [inline x-pack/filebeat/input/awss3/sqs_s3_event.go:205]: Thanks for the PR! Sorry I know it's still a draft, I just want to put a note here. Could we add a debug message here to record we changed the visibility timeout based on the config param please?

File: x-pack/filebeat/input/awss3/sqs_s3_event.go (hunk 1 above)
Comment (faec) [reply x-pack/filebeat/input/awss3/sqs_s3_event.go:214]: This whole thing is wrapped in a for loop, so once the timer expires it'll call the same renewal code above, the only difference should be that this also calls it before the first select. (Does that answer your question? I'm not sure I followed what you meant.)

File: x-pack/filebeat/input/awss3/sqs_s3_event.go (hunk 2 above)
Comment (faec) [reply x-pack/filebeat/input/awss3/sqs_s3_event.go:208]: Done

File: x-pack/filebeat/input/awss3/sqs_s3_event.go (hunk 1 above)
Comment (kaiyan-sheng) [reply x-pack/filebeat/input/awss3/sqs_s3_event.go:214]: oh yes...!!! Missed the for loop!! Thanks!!

**Review by kaiyan-sheng — APPROVED** [review]

**Comment by andrewkroh** [comment]
> The visibility_timeout flag in the SQS input isn't initialized correctly. It is used when extending visibility timeouts, but not when initially fetching the message

https://github.com/elastic/beats/blob/e3a8223d43bd3af01183c4f559280f1da72007de/x-pack/filebeat/input/awss3/interfaces.go#L141-L144

I thought that's what this code was doing during the initial `ReceiveMessage`. Or are you saying that `a.visibilityTimeout` isn't initialized?

**Pull request closed without merging**